keytree/src/lib.rs

```diff
@@ -148,21 +149,23 @@ mod tests {
     use rand::SeedableRng;
     use rand_chacha::ChaChaRng;
 
+    use hex;
+
     #[test]
     fn random_xprv_test() {
         let seed = [0u8; 32];
         let mut rng = ChaChaRng::from_seed(seed);
         let xprv = Xprv::random(&mut rng);
 
         // the following are hard-coded based on the previous seed
-        let expected_dk = [
-            159, 7, 231, 190, 85, 81, 56, 122, 152, 186, 151, 124, 115, 45, 8, 13, 203, 15, 41,
-            160, 72, 227, 101, 105, 18, 198, 83, 62, 50, 238, 122, 237,
-        ];
-        let expected_scalar = Scalar::from_bits([
-            74, 83, 195, 251, 188, 89, 151, 14, 229, 248, 90, 248, 19, 135, 93, 255, 193, 58, 144,
-            74, 46, 83, 174, 126, 101, 250, 13, 234, 110, 98, 201, 1,
-        ]);
+        let dk_hex = "9f07e7be5551387a98ba977c732d080dcb0f29a048e3656912c6533e32ee7aed";
+        let scalar_hex = "4a53c3fbbc59970ee5f85af813875dffc13a904a2e53ae7e65fa0dea6e62c901";
+
+        let expected_dk = hex::decode(dk_hex).unwrap();
```

Comment:
Generally, in tests one does not need to decode from hex, only encode.

Why not writing something like this for brevity:

```rust
assert_eq!(
    hex::encode(&xprv.dk),
    "9f07e7be5551387a98ba977c732d080dcb0f29a048e3656912c6533e32ee7aed"
);
assert_eq!(
    hex::encode(xprv.scalar.as_bytes()),
    "4a53c3fbbc59970ee5f85af813875dffc13a904a2e53ae7e65fa0dea6e62c901"
);
```

Reply:
> Why not writing something like this

I took some pains in this PR to try to make all of the assert statements as short and obvious as possible (i.e. always comparing exactly the thing I was testing). But you're right, that makes these tests very long.